src/tarantool.client/ResponseReader.cs

```diff
@@ -29,14 +32,49 @@ internal class ResponseReader : IResponseReader
 
         private bool _disposed;
 
-        public ResponseReader(ILogicalConnection logicalConnection, ClientOptions clientOptions, INetworkStreamPhysicalConnection physicalConnection)
+        public ResponseReader(ClientOptions clientOptions, INetworkStreamPhysicalConnection physicalConnection)
         {
             _physicalConnection = physicalConnection;
-            _logicalConnection = logicalConnection;
             _clientOptions = clientOptions;
             _buffer = new byte[clientOptions.ConnectionOptions.ReadStreamBufferSize];
         }
 
+        public Task<MemoryStream> GetResponseTask(RequestId requestId)
+        {
+            lock (this)
```

Comment:
Better make _pendingRequests not readonly and set it to null in Dispose via InterlockedExchange.

Then you can copy it here via Interlocked.Exchange to local variable and check for null-ness.

Then we can rid of _disposed flag or made it calculatable.